"Other models" section (#9)
greenelab/biopriors-review: content/05.other-models.md

## Other models incorporating biological structure

> Knowledge about biological entities can commonly be organized in an ontology, which is a hierarchy or directed graph that encodes the relationships between entities.

Reviewer: ```suggestion
Knowledge about biological entities is often organized in an ontology, which is a directed graph that encodes the relationships between entities.
```

> The Gene Ontology (GO) [@doi:10.1093/nar/gky1055] is one example of such a structure, describing the relationships between cellular subsystems and other attributes describing proteins or genes.

Reviewer: ```suggestion
The Gene Ontology (GO) [@doi:10.1093/nar/gky1055] describes the relationships between cellular subsystems and other attributes describing proteins or genes.
```

DCell [@doi:10.1038/nmeth.4627] uses GO to inform the connectivity of a neural network predicting the effects of gene deletions on yeast growth.
The authors show that DCell performs comparably to an unconstrained neural network for this task.
Additionally, it has the advantage of interpretability: a cellular subsystem with high neuron output values under a particular gene deletion can be inferred to be strongly affected by the gene deletion, providing a putative genotype-phenotype association.
DeepGO [@doi:10.1093/bioinformatics/btx624] uses a similar approach to predict protein function from amino acid sequence, using a neural network constrained by the dependencies of GO.

> However, a follow-up paper by the same authors [@doi:10.1101/615260] showed that this hierarchy-aware approach can be outperformed by a hierarchy-naive CNN, which uses only raw amino acid sequence and similarity to labeled proteins in the training set.

Reviewer: This is interesting - did they evaluate on proteins that have very little homology to other proteins in the training set? I wonder if those would be more aided by a constrained approach. Just a random note - no suggested changes to text :)

Author: They don't segment their results by sequence homology or talk much about it in the most recent paper, so it's hard to tell.

Intuitively, I'd think what you're saying is likely to be true, since the CNN model is just using sequence information (the earlier DeepGO model also uses protein interaction information, which they argue isn't that helpful since it isn't available for many proteins with unknown function).

They evaluated their model on the CAFA3 evaluation set - not sure what the distribution of sequence homology is like for those proteins (maybe you know?)
